File: regexlib/javascript_test_file/regexlib_5965.js
/* 5965
 * <\s*/?\s*\w+(\s*\w+\s*=\s*(['"][^'"]*['"]|[\w#]+))*\s*/?\s*>
 * EXPONENT
 * nums:3
 * EXPONENT AttackString:"<1"+"000="*32+"! _1_EOA(i or ii)"
 */
var REGEX = new RegExp("<\s*/?\s*\w+(\s*\w+\s*=\s*(['"][^'"]*['"]|[\w#]+))*\s*/?\s*>")
for(var i = 1; i <= 5000000; i++) {
    var time = Date.now();
    var attack_str = '<1' + '000='.repeat(i*1) + '! _1_EOA(i or ii)'
    REGEX.exec(attack_str)
    // REGEX.test(attack_str);
    // attack_str.search(REGEX)
    // attack_str.match(REGEX)
    var time_cost = Date.now() - time;
    console.log(i * 1 + ": " + time_cost+" ms")
}

File: regexlib/2021-5-15/python_re_test_file/regexlib_5965.py
# 5965
# <\s*/?\s*\w+(\s*\w+\s*=\s*(['"][^'"]*['"]|[\w#]+))*\s*/?\s*>
# EXPONENT
# nums:4
# EXPONENT AttackString:"<1"+"000="*32+"! _1_EOA(i or ii)"

import re
from time import perf_counter

regex = """<\s*/?\s*\w+(\s*\w+\s*=\s*(['"][^'"]*['"]|[\w#]+))*\s*/?\s*>"""
REGEX = re.compile(regex)
for i in range(0, 150000):
    ATTACK = "<1" + "000=" * i * 1 + "! _1_EOA(i or ii)"
    LEN = len(ATTACK)
    BEGIN = perf_counter()
    m = REGEX.search(ATTACK)
    # m = REGEX.match(ATTACK)
    DURATION = perf_counter() - BEGIN
    print(f"{i *1}: took {DURATION} seconds!")